controllers/datamovement_controller.go

```diff
@@ -192,62 +218,175 @@ func (r *DataMovementReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, handleInvalidError(err)
 	}
 
+	// Expand the context with cancel and store it in the map so the cancel function can be used in
+	// another reconciler loop. Also add NamespacedName so we can retrieve the resource.
+	ctxCancel, cancel := context.WithCancel(ctx)
+	r.contexts.Store(dm.Name, dataMovementCancelContext{
+		ctx:    ctxCancel,
+		cancel: cancel,
+	})
+
+	// Pull configurable values from the ConfigMap
+	configMap := &corev1.ConfigMap{}
+	if err := r.Get(ctx, types.NamespacedName{Name: configMapName, Namespace: configMapNamespace}, configMap); err != nil {
+		log.Info("Config map not found - requeueing")
+		return ctrl.Result{}, err
+	}
+	log.Info("Config map found:", "configMap", configMap, "data", configMap.Data)
```

Comment:
Does logging the `configMap` also log the `data`? Or is the configMap converted to a namespace/name value?

Reply:
The latter: 
```
2022-10-19T15:38:53.785-0500	INFO	Config map found:	{"controller": "nnfdatamovement", "controllerGroup": "nnf.cray.hpe.com", "controllerKind": "NnfDataMovement", "NnfDataMovement": {"name":"dm-test","namespace":"default"}, "namespace": "default", "name": "dm-test", "reconcileID": "2a1fcc91-8a47-4e27-808f-f7746418258b", "configMap": {"apiVersion": "v1", "kind": "ConfigMap", "namespace": "default", "name": "nnf-dm-config"}, "data": {"dcpProgressInterval":"1s","dmCommand":"sleep 5","dmNumProcesses":"","dmProgressInterval":"1s"}}
```

Reply:
I'm not sure "configMap" buys us anything. What else would it be?

Reply:
I'll remove.